app/controllers/repositories_controller.rb

```diff
@@ -100,14 +103,21 @@ def branches
   end
 
   def notify_push
-    gitlab_event = request.headers['X-Gitlab-Event']
-    if gitlab_event.nil? || !gitlab_event.end_with?('Push Hook')
-      return render nothing: true, status: :unprocessable_entity
+    hook_info = Webhooks::GitLab.new(request, @repository)
+    if hook_info.valid_request?
+      if hook_info.valid_address?
+        if hook_info.valid_branch?
+          @repository.cancel_processing_of_repository unless %w(READY ERROR).include? @repository.last_processing_state
```

Comment:
Yes, I got the same impression, but since the issue is only about the webhooks I chose to keep it as it was.